src/router/gallery.js

```diff
@@ -210,56 +256,158 @@ module.exports = function createGalleryRouter(config) {
 					buffer: buffer
 				};
 
+				// note: posthooks are handled in this function
 				upload({
 					req: req,
 					res: res,
 					data: data
 				});
 			});
 		}
-	});
+	}
+
+	var postParams = ["/"];
+
+	if (config.preHooks.post) {
+		if (typeof config.preHooks.post === "function") {
+			postParams.push(config.preHooks.post);
+		} else if (config.preHooks.post instanceof Array) {
+			postParams = postParams.concat(config.preHooks.post);
+		} else {
+			throw new Error("config.preHooks.post must be a function or Array");
+		}
+	}
+
+	postParams.push(post);
+	router.post.apply(router, postParams);
 
-	router.get("/:id", function(req, res) {
-		var filter = createFilterObjFromParams({
-			belongsTo: config.belongsTo,
-			params: req.params
-		});
 
+	/*
+		 ██████  ███████ ████████  ██████  ███    ██ ███████
+		██       ██         ██    ██    ██ ████   ██ ██
+		██   ███ █████      ██    ██    ██ ██ ██  ██ █████
+		██    ██ ██         ██    ██    ██ ██  ██ ██ ██
+		 ██████  ███████    ██     ██████  ██   ████ ███████
+	*/
+
+	function getOne(req, res) {
 		var id = req.params.id;
-		infoProxy.readOneById(id, filter, createResponseHandler(req, res));
-	});
 
-	router.put("/:id", function(req, res) {
-		var filter = createFilterObjFromParams({
-			belongsTo: config.belongsTo,
-			params: req.params
-		});
+		// avoid accidentally apply id to the filter from preHooks
```

Comment:
például a /:APIkey/:belongsTo/:id esetén az id is automatikusan a filter-re kerül. Ezt lehet kezelni a prehookban is, ez csak egy extra réteg biztonság.